### PR TITLE
Set fileversion created date correctly

### LINKED ISSUE #1146: Media new FileVersion created date is false
Every fileversion have the same created date in history

### PR DESCRIPTION
**Pull request by alexander-schranz**
__Tasks:__

- [x] test coverage
- [x] gather feedback for my changes
- [x] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | no
| Fixed tickets | fixes #1146 
| BC Breaks     | none
| Doc           | none

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php [hunk 1]
```diff
@@ -637,9 +638,6 @@ protected function setDataToMedia(Media $media, $data, $user)
                     case 'url':
                         $media->setUrl($value);
                         break;
-                    case 'versionUrls':
-                        $media->setVersionUrls($value);
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php]: this function was removed in #932 from media object

File: src/Sulu/Bundle/MediaBundle/Api/Media.php [hunk 2]
```diff
@@ -622,6 +622,20 @@ public function setUrl($url)
...
+     *
+     * @return $this
+     */
+    public function setChanged($changed)
+    {
+        if (is_string($changed)) {
+            $changed = new \DateTime($changed);
+        }
+        $this->getFileVersion()->setChanged($changed);
+        return $this;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: before return there should be an empty line!

File: src/Sulu/Bundle/MediaBundle/Api/Media.php [hunk 3]
```diff
@@ -668,7 +682,7 @@ public function getChanger()
      */
     public function getCreated()
     {
-        return $this->getFileVersion()->getCreated();
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: we should use the same method to get the entity in all functions here... sometimes it uses `$this->getFileVersion()` and the others uses `$this->entity` could you change it to the same in this PR?

File: src/Sulu/Bundle/MediaBundle/Entity/File.php [hunk 4]
```diff
@@ -86,6 +86,19 @@ public function getChanged()
...
+     * Set changed.
+     *
+     * @param \DateTime $changed
+     *
+     * @return $this
+     */
+    public function setChanged($changed)
+    {
+        $this->changed = $changed;
+        return $this;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Entity/File.php]: empty line before return

File: src/Sulu/Bundle/MediaBundle/Entity/FileVersion.php [hunk 5]
```diff
@@ -270,6 +275,19 @@ public function getCreated()
...
+     * Get created.
+     *
+     * @param \DateTime $created
+     *
+     * @return $this
+     */
+    public function setCreated($created)
+    {
+        $this->created = $created;
+        return $this;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Entity/FileVersion.php]: same here

File: src/Sulu/Bundle/MediaBundle/Entity/FileVersion.php [hunk 6]
```diff
@@ -280,6 +298,19 @@ public function getChanged()
...
+     * Set changed.
+     *
+     * @param \DateTime $changed
+     *
+     * @return $this
+     */
+    public function setChanged($changed)
+    {
+        $this->changed = $changed;
+        return $this;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Entity/FileVersion.php]: and here(=

**Comment by wachterjohannes** [comment]
@chirimoya could you please test it? the code is ok (=

File: src/Sulu/Bundle/MediaBundle/Api/Media.php (hunk 3 above)
Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: its the special Media API object 
You dont wanted the created date of the FileVersion there you want the created date of the Media

Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: The created date for a specific fileVersions are under 'versions' http://sulu.lo/admin/api/media/1

Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Api/Media.php]: ok then its ok (= i haven´t seen the difference before!

**Comment by chirimoya** [comment]
@alexander-schranz ... tests are failing, can you please check?

**Pull request merged**